### DISCUSSION TITLE
Converters with nested models

### DISCUSSION THREAD

**Discussion by johnmcgowan1** [lwtnn/lwtnn#148] [Q&A]
Hi again lwtnn experts,

Starting another discussion as I expect someone here may have addressed this issue before. I would like to use lwtnn for a couple nn models in my atlas analysis. One is a pretty simple densely connected network. Here lwtnn works great and is saving me a lot of time and effort. The second network is built off the first, in that the first is imported and in keras functional api I do something like 

l1 = imported_model(input)
l2 = dense(l1)
etc.It 

This way I can easily import the first model and freeze and unfreeze those weights as needed. However this presents an issue when I am converting my architecture into the variables file used to make the json to construct the lwtnn file. The node class can't handle this layer which is actually a model. I think this is because the 'inbound_node' has the structure [[['name',1,0]]] instead of [[['name',0,0]]] and I get an error when trying to make a node with idx 1. I don't have much an understanding of how models are codified into json, I'd expect my inbound nodes to resemble the size of inputs being passed to it, i.e. the 1 would come from the output of the model layer being a single value, but seems to not be the case since every other inbound node in my jsons has the format [[['name',0,0]]]. 

Has anyone dealt with a workaround of this before? As always, any help is much appreciated.

Best,
John

**Comment by johnmcgowan1** [comment]
[adv_even_0713.txt](https://github.com/lwtnn/lwtnn/files/6886368/adv_even_0713.txt)

Attaching the architecture json if it is helpful for anyone interested.

**Comment by dguest** [comment]
Unfortunately we don't currently support what you're asking for. It's not theoretically difficult to do (we just have to teach the converter to navigate around the "model") but it would probably take a bit of experimentation to get it right. I can't promise I can look at this very soon, but if you attach the whole model (with the weights) and point to the code that generates it it might help a bit.

**Reply by johnmcgowan1** [reply]
Hi @dguest. Thank you for the follow up and apologies for my late response. It seems I can't attach json or h5 files, so I've made them available via this public link: https://cernbox.cern.ch/index.php/s/coZO5BaAvpo5ODO. The code to make the models is stored in this branch here: https://gitlab.cern.ch/jmcgowan/ann_studies/-/tree/cedar_development. Its a little complicated. I make one model in the train_classifier.py and this ends up being the model that is nested in the larger model, which is made in train_adversary_mse.py. Then they are trained together in adve_training_mse.py

I think I may be able to come up with a workaround on the making the model side. If not I'll try some experimentation myself with the existing scripts.
